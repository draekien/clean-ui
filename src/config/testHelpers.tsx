import React, { ReactNode } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: ReactNode) =>
  render(<ThemeProvider>{element}</ThemeProvider>);
