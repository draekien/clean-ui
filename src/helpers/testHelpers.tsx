import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: React.ReactNode) => {
  return render(<ThemeProvider>{element}</ThemeProvider>);
};

export const hexToRgb = (hex: string) => {
  hex = hex.replace(/[^0-9A-F]/gi, '');
  const bigint = parseInt(hex, 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
};
