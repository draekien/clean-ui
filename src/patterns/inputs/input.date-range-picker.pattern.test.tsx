import React from 'react';
import { DateRangePickerInput } from '../..';
import { themedRender } from '../../helpers/testHelpers';

describe('Date range picker pattern', () => {
  it('should match snapshot', () => {
    const { container } = themedRender(
      <DateRangePickerInput label="Dates" inputId="dates" />
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when dates are populated', () => {
    const { container } = themedRender(
      <DateRangePickerInput
        label="Dates"
        inputId="dates"
        dateFrom={new Date(2021, 1, 1)}
        dateTo={new Date(2021, 1, 2)}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
