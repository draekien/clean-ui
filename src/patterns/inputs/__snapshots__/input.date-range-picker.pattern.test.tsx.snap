// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date range picker pattern should match snapshot 1`] = `
<div>
  <div>
    <div
      class="css-1iq9e3v-DateRangePickerInput"
    >
      <div
        class="css-109th3q-InputWrapper"
      >
        <div
          class="css-1qm0f6g-InputLabel"
        >
          <label
            for="dates"
          >
            <div
              class="css-14i3diu-Text css-vurnku"
            >
              Dates
            </div>
          </label>
        </div>
        <div
          class="css-xcsx4z"
        >
          <input
            class="css-msl6v2"
            id="dates"
            type="text"
            value="__/__/____ - __/__/____"
          />
          <div
            class="css-e7011r"
          >
            <button
              class="css-1ipglsu-Button"
              icon="calendar_today"
            >
              <span
                class="css-zohjwb-ButtonIcon"
              >
                <i
                  class="material-icons  css-zif8ik-Icon"
                >
                  calendar_today
                </i>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
